carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/router/HttpMethodInfo.java

```diff
@@ -72,39 +62,52 @@
         this.args = args;
     }
 
+    HttpMethodInfo(Method method, Object handler, HttpRequest request,
+                   HttpResponder responder, Object[] args,
+                   ExceptionHandler exceptionHandler, String mediaType,
+                   HttpStreaming httpStreaming) throws HandlerException {
+        this(method, handler, request, responder, args, exceptionHandler, mediaType);
+
+        if (!method.getReturnType().equals(Void.TYPE)) {
+            throw new HandlerException(HttpResponseStatus.INTERNAL_SERVER_ERROR,
+                    "Resource method should be void if it accepts chunked requests");
```

Comment:
Better to have this like,  if (!(Void.TYPE).equals(method.getReturnType())) to avoid possible NPE.